### PR TITLE
Step indicator update

### PR DESCRIPTION
**Pull request by dakota002**
This introduces the step process for the user. Once logged in, they can access these steps from the user page.

### CODE REVIEW

**Comment by lpsinger** [comment]
Would you please add some screen shots?

**Review by lpsinger — CHANGES_REQUESTED** [review]
I'm a bit confused about how this works. Where is `setActiveSteps` defined?

**Comment by dakota002** [comment]
Screen Shot 2022-06-30 at 12 43 50 PM
Screen Shot 2022-06-30 at 12 44 03 PM
Screen Shot 2022-06-30 at 12 44 39 PM
Screen Shot 2022-06-30 at 12 44 48 PM

**Comment by dakota002** [comment]
Screen Shot 2022-06-30 at 12 44 39 PM

**Comment by dakota002** [comment]
Screen Shot 2022-06-30 at 12 44 48 PM

File: app/routes/user/streaming_steps/credentials.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  const client_credentials = await machine.getClientCredentials()
+  return { client_credentials }
+}
+
+export default function Credentials() {
+  //Data passed between steps
+  const clientData = useClient()
+  // Data loaded for options
+  const { client_credentials } =
+    useLoaderData<Awaited<ReturnType<typeof loader>>>() ?? []
```
Comment (lpsinger) [inline app/routes/user/streaming_steps/credentials.tsx:23]: Why would this ever return null?

File: app/components/ClientCredentialForm.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  setItems: React.Dispatch<React.SetStateAction<ClientCredentialData[]>>
+}) {
+  const { client_credentials } =
+    useLoaderData<Awaited<ReturnType<typeof loader>>>() ?? []
+  const [items] = useState<ClientCredentialData[]>(client_credentials)
+  const defaultName = ''
+  const [name, setName] = useState(defaultName)
+  const defaultScope = 'gcn.nasa.gov/kafka-public-consumer'
+  const [scope, setScope] = useState(defaultScope)
+  const siteKey = 'site-key'
```
Comment (lpsinger) [inline app/components/ClientCredentialForm.tsx:27]: Would you please restore the logic from main that gets this from the environment?

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/user/streaming_steps/credentials.tsx (hunk 1 above)
Comment (dakota002) [reply app/routes/user/streaming_steps/credentials.tsx:23]: It shouldn't, I believe that was from testing. Fixing now

File: app/components/ClientCredentialForm.tsx (hunk 2 above)
Comment (dakota002) [reply app/components/ClientCredentialForm.tsx:27]: Updated!

**Review by lpsinger — APPROVED** [review]

**Review by lpsinger — CHANGES_REQUESTED** [review]
Looks great, but needs to be rebased.

**Review by lpsinger — APPROVED** [review]
I rebased it.

**Pull request merged**